pkg/descheduler/policyconfig.go

```diff
@@ -45,26 +46,125 @@ func LoadPolicyConfig(policyConfigFile string, client clientset.Interface, regis
 		return nil, fmt.Errorf("failed to read policy config file %q: %+v", policyConfigFile, err)
 	}
 
+	internalPolicy, err := decode(policyConfigFile, policy, client, registry)
+	if err != nil {
+		return nil, err
+	}
+	return internalPolicy, nil
+}
+
+func decode(policyConfigFile string, policy []byte, client clientset.Interface, registry pluginregistry.Registry) (*api.DeschedulerPolicy, error) {
 	versionedPolicy := &v1alpha1.DeschedulerPolicy{}
+	internalPolicy := &api.DeschedulerPolicy{}
+	var err error
 
-	decoder := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion)
+	decoder := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion, v1alpha2.SchemeGroupVersion, api.SchemeGroupVersion)
 	if err := runtime.DecodeInto(decoder, policy, versionedPolicy); err != nil {
-		return nil, fmt.Errorf("failed decoding descheduler's policy config %q: %v", policyConfigFile, err)
+		if err.Error() == "converting (v1alpha2.DeschedulerPolicy) to (v1alpha1.DeschedulerPolicy): unknown conversion" {
+			klog.V(1).InfoS("Tried reading v1alpha2.DeschedulerPolicy and failed. Trying legacy conversion now.")
+		} else {
+			return nil, fmt.Errorf("failed decoding descheduler's policy config %q: %v", policyConfigFile, err)
+		}
 	}
-
-	// Build profiles
-	internalPolicy, err := V1alpha1ToInternal(client, versionedPolicy, registry)
-	if err != nil {
-		return nil, fmt.Errorf("failed converting versioned policy to internal policy version: %v", err)
+	if versionedPolicy.APIVersion == "descheduler/v1alpha1" {
+		// Build profiles
+		internalPolicy, err = V1alpha1ToInternal(client, versionedPolicy, registry)
+		if err != nil {
+			return nil, fmt.Errorf("failed converting versioned policy to internal policy version: %v", err)
+		}
+		err = validateDeschedulerConfiguration(*internalPolicy, registry)
```

Comment:
the calls to `validateDeschedulerConfiguration` can be moved out of this if/else since they are both on parsed `internalPolicy` structs right?